internal/services/qumulo/qumulo_file_system_resource.go

```diff
@@ -0,0 +1,274 @@
+package qumulo
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/qumulostorage/2022-10-12/filesystems"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/qumulo/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = FileSystemResource{}
+var _ sdk.ResourceWithUpdate = FileSystemResource{}
+
+const (
+	offerId     = "qumulo-saas-mpp"
+	publisherId = "qumulo1584033880660"
+)
+
+type FileSystemResource struct{}
+
+func (r FileSystemResource) ModelObject() interface{} {
+	return &FileSystemResourceSchema{}
+}
+
+type FileSystemResourceSchema struct {
+	AvailabilityZone  string                 `tfschema:"availability_zone"`
+	AdminPassword     string                 `tfschema:"admin_password"`
+	DelegatedSubnetId string                 `tfschema:"delegated_subnet_id"`
+	InitialCapacity   int64                  `tfschema:"initial_capacity"`
+	Location          string                 `tfschema:"location"`
+	MarketplacePlanId string                 `tfschema:"marketplace_plan_id"`
+	Name              string                 `tfschema:"name"`
+	ResourceGroupName string                 `tfschema:"resource_group_name"`
+	StorageSku        string                 `tfschema:"storage_sku"`
+	Tags              map[string]interface{} `tfschema:"tags"`
+	UserEmailAddress  string                 `tfschema:"user_email_address"`
+}
+
+func (r FileSystemResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return filesystems.ValidateFileSystemID
+}
+
+func (r FileSystemResource) ResourceType() string {
+	return "azurerm_qumulo_file_system"
+}
+
+func (r FileSystemResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: validate.FileSystemName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"admin_password": {
+			Type:      pluginsdk.TypeString,
+			Required:  true,
+			ForceNew:  true,
+			Sensitive: true,
+		},
+
+		"delegated_subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
```

Comment:
presumably the `subnet` would need to be delegated to be usable by Qumulo (e.g. have nothing else in it/be delegated to Qumulo) - so we could probably remove `delegated_` from the start of this and just make this `subnet_id` to match other resources?